### DISCUSSION TITLE
Disable status page publishing

### DISCUSSION THREAD

**Discussion by bradyjoslin** [upptime/upptime#107] [Q&A] [answered]
From the [getting started docs](https://upptime.js.org/docs/get-started):

> If you don't want to have a publicly-available status website, you don't have to enable publishing.

Created a new private repo based on the template, updated the config file, did not enable GitHub Pages, but a status site was published publicly automatically anyway.  This appears to be done by maxheld83/ghpages in the site.yml.

Is there another supported way to disable publishing a public status page for cases where it is not desired?

**Comment by AnandChowdhary** [comment]
Quick question, have you tried disabling GitHub Pages from the settings? Does it auto re-enable when you manually run "Static Site CI"?

**Comment by bradyjoslin** [comment]
When I initially created the repo based on the template, I checked the box to `Include all branches`, which led to the initial publish.  To disable I deleted the `gh-pages` branch then set the GitHub Pages setting to None.  But, rerunning "static site ci" republished the status page.

What appears to be happening is that GitHub automatically publishes GitHub Pages when a `gh-pages` branch is present.  I just tried it out on a vanilla private repo and confirmed that is what's happening, not sure if there's a way to turn that off.  

Based on these findings, a workaround I did for my upptime is to create an empty `gh-pages` branch and added branch protection to disallow pushing to that branch, enabling the option for the rule to be enforced for administrators.  With that repo configuration, the site CI process fails since it can't successfully push to that branch.  

It would be nice if there was a setting which would skip the step that pushes to the `gh-pages` branch completely.

**Comment by AnandChowdhary** [comment, marked as answer]
Aah, that makes sense. For now, what I've done is that if you have a private repository, we don't generate the site, we just generate an `index.html` page that says "404" and put that in the dist directory. This way, even if it pushed to `gh-pages` and the site is deployed, there's nothing to see there. Soon, I want to add the feature to completely disable publishing if it's a private repo rather than this hack.

For people who have a private repo and still want to publish their status website, they now have to set this:

```yaml
status-website:
  publish: true
```
